File: microservices/cinema_service/project/routes/movie_routes.py
from flask import Blueprint

from ..controllers.movie_controller import (
    get_seat_map,
    list_movies,
    list_showtimes,
    lock_seats,
    movie_details,
    release_seats,
)


movie_bp = Blueprint("movie_bp", __name__)

movie_bp.route("/movies", methods=["GET"])(list_movies)
movie_bp.route("/movies/<int:movie_id>", methods=["GET"])(movie_details)
movie_bp.route("/movies/<int:movie_id>/showtimes", methods=["GET"])(list_showtimes)
movie_bp.route("/showtimes/<int:showtime_id>/seats", methods=["GET"])(get_seat_map)
movie_bp.route("/showtimes/<int:showtime_id>/seats/lock", methods=["POST"])(lock_seats)
movie_bp.route("/showtimes/<int:showtime_id>/seats/release", methods=["POST"])(release_seats)

File: microservices C#/cinema_service/Routes/MovieRoutes.cs
using CinemaService.Controllers;
using CinemaService.Models;

namespace CinemaService.Routes;

public static class MovieRoutes
{
    public static void MapMovieRoutes(this WebApplication app)
    {
        app.MapGet("/", (MovieController controller) => controller.Index());
        app.MapGet("/api/movies", (MovieController controller) => controller.ListMovies());
        app.MapGet("/api/movies/{movieId:int}", (MovieController controller, int movieId) => controller.MovieDetails(movieId));
        app.MapGet("/api/movies/{movieId:int}/showtimes", (MovieController controller, int movieId, string? date) => controller.ListShowtimes(movieId, date));
        app.MapGet("/api/showtimes/{showtimeId:int}/seats", (MovieController controller, int showtimeId, System.Security.Claims.ClaimsPrincipal principal) => controller.GetSeatMap(showtimeId, principal)).RequireAuthorization();
        app.MapPost("/api/showtimes/{showtimeId:int}/seats/lock", (MovieController controller, int showtimeId, SeatCodesRequest request, System.Security.Claims.ClaimsPrincipal principal) => controller.LockSeats(showtimeId, request, principal)).RequireAuthorization();
        app.MapPost("/api/showtimes/{showtimeId:int}/seats/release", (MovieController controller, int showtimeId, SeatCodesRequest request, System.Security.Claims.ClaimsPrincipal principal) => controller.ReleaseSeats(showtimeId, request, principal)).RequireAuthorization();
    }
}
